store/generated_key/memstore/v2/memstore.go

```diff
@@ -0,0 +1,213 @@
+// since they'd resolve to the same commitment and blob key. This shouldn't matter
+// given this is typically used for testing standard E2E functionality against a rollup
+// stack which SHOULD never submit an identical batch more than once.
+func unsafeRandomBytes(size uint) []byte {
+	entropy := make([]byte, size)
+	_, _ = rand.Read(entropy)
+	return entropy
+}
+
+func unsafeRandInt(maxValue int64) *big.Int {
+	randInt, _ := rand.Int(rand.Reader, big.NewInt(maxValue))
+	return randInt
+}
+
+func unsafeRandUint32() uint32 {
+	// #nosec G115 - downcasting only on random value
+	return uint32(unsafeRandInt(32).Uint64())
+}
+
+/*
+MemStore is a simple in-memory store for blobs which uses an expiration
+time to evict blobs to best emulate the ephemeral nature of blobs dispersed to
+EigenDA V2 operators.
+*/
+type MemStore struct {
+	// keccak(RLP(randomlyGeneratedCert)) -> Blob
+	*ephemeraldb.DB
+	log logging.Logger
+
+	g1SRS []bn254.G1Affine
+	codec codecs.BlobCodec
+}
+
+var _ common.GeneratedKeyStore = (*MemStore)(nil)
+
+// New ... constructor
+func New(
+	ctx context.Context, log logging.Logger, config *memconfig.SafeConfig,
+	g1SRS []bn254.G1Affine,
+) (*MemStore, error) {
+	return &MemStore{
+		ephemeraldb.New(ctx, config, log),
+		log,
+		g1SRS,
+		codecs.NewIFFTCodec(codecs.NewDefaultBlobCodec()),
+	}, nil
+}
+
+// generateRandomCert ... generates a pseudo random EigenDA V2 certificate
+func (e *MemStore) generateRandomCert(blobContents []byte) (*verification.EigenDACert, error) {
+	// compute kzg data commitment. this is useful for testing
+	// READPREIMAGE functionality in the arbitrum x eigenda integration since
+	// preimage key is computed within the VM from hashing a recomputation of the data
+	// commitment
+	dataCommitment, err := verification.GenerateBlobCommitment(e.g1SRS, blobContents)
+	if err != nil {
+		return nil, err
+	}
+
+	x := dataCommitment.X.BigInt(&big.Int{})
+	y := dataCommitment.Y.BigInt(&big.Int{})
+
+	g1CommitPoint := cert_verifier_binding.BN254G1Point{
+		X: x,
+		Y: y,
+	}
+
+	pseudoRandomBlobInclusionInfo := cert_verifier_binding.BlobInclusionInfo{
+		BlobCertificate: cert_verifier_binding.BlobCertificate{
+			BlobHeader: cert_verifier_binding.BlobHeaderV2{
+				Version:       0,                            // only supported version as of now
+				QuorumNumbers: []byte{byte(0x0), byte(0x1)}, // quorum 0 && quorum 1
+				Commitment: cert_verifier_binding.BlobCommitment{
+					LengthCommitment: cert_verifier_binding.BN254G2Point{
+						X: [2]*big.Int{unsafeRandInt(1000), unsafeRandInt(1000)},
+						Y: [2]*big.Int{unsafeRandInt(1000), unsafeRandInt(1000)},
+					},
+					LengthProof: cert_verifier_binding.BN254G2Point{
+						X: [2]*big.Int{unsafeRandInt(1), unsafeRandInt(1)},
+						Y: [2]*big.Int{unsafeRandInt(1), unsafeRandInt(1)},
+					},
+					Commitment: g1CommitPoint,
+					// #nosec G115 - can never overflow on 16MiB blobs
+					Length: uint32(len(blobContents)),
```

Comment:
Let's change this to be the # of field elements instead

Reply:
[9c714bd](https://github.com/Layr-Labs/eigenda-proxy/pull/321/commits/9c714bd88aa56f25e3fc64a02ee69eb2b3b6a856)